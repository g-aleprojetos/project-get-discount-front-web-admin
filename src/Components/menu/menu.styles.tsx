import styled from 'styled-components';
import colors from 'resources/colors';

export interface PropsMenu {
  openMenu?: boolean;
  backgroundDark?: boolean;
  isActive?: boolean;
  idiomaSelecionado?: boolean;
}

export const ContainerNav = styled.nav`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 89px;
  padding-right: 55px;
`;

export const ContainerContent = styled.ul`
  display: flex;
  align-items: center;
  flex-flow: row nowrap;
  position: static;
  background-color: 'transparent';
  transform: 'none';
  top: 'auto';
  right: 'auto';
  width: 'auto';
  height: auto;
  padding-top: 'auto';
  transition: 'none';
`;

export const Item = styled.li<PropsMenu>`
  display: flex;
  justify-content: center;
  padding: 8px;
  margin: 5px 21px;
  color: ${colors.white};
`;

export const ItemNav = styled(Item)`
  width: 100%;
  white-space: nowrap;
  cursor: ${props => (props.isActive ? 'default' : 'pointer')};
  :hover {
    background-color: 'none';
  }
`;
