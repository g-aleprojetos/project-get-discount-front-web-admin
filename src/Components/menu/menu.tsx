import React, {useState} from 'react';
import {useNavigate} from 'react-router-dom';
import {TextH2} from 'Components/text';
import {useAuthenticatorContext} from 'context/authetication';
import {Roles} from 'types/enuns';
import colors from 'resources/colors';
import rotas from 'types/rotasAPP';
import * as S from './menu.styles';

export type Props = S.PropsMenu & {
  handleLogout: () => void;
};

export const Menu = (props: Props) => {
  const {handleLogout} = props;
  const navigate = useNavigate();
  const {usuario} = useAuthenticatorContext();
  const [telaAtiva, setTelaAtiva] = useState<string>(rotas.AdminPage);

  const handleNavegate = (item: string) => {
    setTelaAtiva(item);
    navigate(item);
  };

  return (
    <S.ContainerNav data-testid={'test_menu'}>
      <S.ContainerContent data-testid={'test_containerContent'}>
        {usuario?.role === Roles.ADMIN && (
          <>
            <S.ItemNav
              isActive={telaAtiva === rotas.AdminPage}
              onClick={() => {
                handleNavegate(rotas.AdminPage);
              }}>
              {telaAtiva === rotas.AdminPage ? (
                <TextH2 peso="bold" cursor="default">
                  P치gina Administrativa
                </TextH2>
              ) : (
                <TextH2 cursor="pointer" cor={colors.lightgray}>
                  P치gina Administrativa
                </TextH2>
              )}
            </S.ItemNav>
            <S.ItemNav
              isActive={telaAtiva === rotas.UserPage}
              onClick={() => {
                handleNavegate(rotas.UserPage);
              }}>
              {telaAtiva === rotas.UserPage ? (
                <TextH2 peso="bold" cursor="default">
                  P치gina Usuario
                </TextH2>
              ) : (
                <TextH2 cursor="pointer" cor={colors.lightgray}>
                  P치gina Usuario
                </TextH2>
              )}
            </S.ItemNav>
          </>
        )}
        <S.ItemNav
          onClick={() => {
            handleLogout();
          }}>
          <TextH2 cursor="pointer" cor={colors.lightgray}>
            Logout
          </TextH2>
        </S.ItemNav>
      </S.ContainerContent>
    </S.ContainerNav>
  );
};
