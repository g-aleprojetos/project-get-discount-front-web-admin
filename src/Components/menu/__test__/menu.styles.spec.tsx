import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../menu.styles';

describe('Input.styles', () => {
  test('ContainerContent DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerContent />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerNav DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerNav />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Item DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Item />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ItemNav DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ItemNav />).toJSON();

    expect(container).toMatchSnapshot();
  });
});
