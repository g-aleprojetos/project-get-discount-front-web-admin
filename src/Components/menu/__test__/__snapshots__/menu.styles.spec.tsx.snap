// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.styles ContainerContent DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  position: static;
  background-color: 'transparent';
  -webkit-transform: 'none';
  -ms-transform: 'none';
  transform: 'none';
  top: 'auto';
  right: 'auto';
  width: 'auto';
  height: auto;
  padding-top: 'auto';
  -webkit-transition: 'none';
  transition: 'none';
}

<ul
  className="c0"
/>
`;

exports[`Input.styles ContainerNav DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 89px;
  padding-right: 55px;
}

<nav
  className="c0"
/>
`;

exports[`Input.styles Item DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px;
  margin: 5px 21px;
  color: #FFFFFF;
}

<li
  className="c0"
/>
`;

exports[`Input.styles ItemNav DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px;
  margin: 5px 21px;
  color: #FFFFFF;
}

.c1 {
  width: 100%;
  white-space: nowrap;
  cursor: pointer;
}

.c1:hover {
  background-color: 'none';
}

<li
  className="c0 c1"
/>
`;
